cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -517,7 +517,12 @@ private static ApplicationDetail makeAppDetail(ApplicationSpecification spec) {
       datasets.add(new DatasetDetail(datasetSpec.getInstanceName(), datasetSpec.getTypeName()));
     }
 
+    // this is only required if there are old apps lying around that failed to get upgrading during
+    // the upgrade to v3.2 for some reason. In those cases artifact id will be null until they re-deploy the app.
+    // in the meantime, we don't want this api call to null pointer exception.
+    ArtifactSummary summary = spec.getArtifactId() == null ?
+      new ArtifactSummary(spec.getName(), null, false) : ArtifactSummary.from(spec.getArtifactId());
```

Comment:
this can never happen for apps created after 3.2, the deployment pipeline requires an artifact in order to create the app spec.